### ISSUE TITLE
Suggestion for a new field PREDEPLOYMENT_CALIB_DATE

### ISSUE THREAD

**Issue by BrianKingNOC** [OneArgo/ArgoVocabs#162]
Our work with sensor makers to ask them to provide calibration information in fully machine-readable json files has shown up that storing information about **calibration date** is uncontrolled and inconsistent.

At present, many DACs include some date information in 
PREDEPLOYMENT_CALIB_COMMENT
by adding a date somewhere in that string, with whatever format they choose for the string. This is not machine parseable.

If sensor makers are going to supply the information and we think it is worth keeping, the cleanest thing would be a new variable
PREDEPLOYMENT_CALIB_DATE(N_PARAM, DATE_TIME)

The predeployment_calib variables are dimensioned on N_PARAM, so the date would be tied back to the sensor serial number identified by 

PARAMETER_SENSOR(N_PARAM, STRING128)

Examples of PARAMETER_SENSOR are CTD_PRES, CTD_TEMP, CTD_CNDC

**Here would be possible manual entries:**

Name

PREDEPLOYMENT_CALIB_DATE

Definition

char 
PREDEPLOYMENT_CALIB_DATE(N_P ARAM, DATE_TIME);
PREDEPLOYMENT_CALIB_DATE:long _name = "Date and time (UTC) of calibration";
PREDEPLOYMENT_CALIB_DATE:conventions = "YYYYMMDDHHMISS"; 
PREDEPLOYMENT_CALIB_DATE:_Fill Value = " ";

Comment

Date-time at which calibration coefficients were determined. If date only is known, pad with zeros for HHMISS. 
Format : YYYYMMDDHHMISS Example : 20010203000000 : Feb 3rd 2001
If calibration date is not known, use "unknown" padded with 7 extra spaces.
If calibration date is not applicable for this parameter, use "not applicable".

**Comment by apswong** [comment]
I agree that we need to create a new metadata variable: PREDEPLOYMENT_CALIB_DATE(N_PARAM, DATE_TIME).

However, for Comment, I would suggest "Calibration date for this parameter from the sensor", not ".... when the coefficients were determined"

And for cases where the date is unknown or not application, the variable should be filled with FillValue, which in this case is " ".  Any other suggested char strings, such as "unknown", "not applicable", etc, are char strings, which do not adhere to the format specification of DATE_TIME.

**Comment by SBS-EREHM** [comment]
FYI, both [RBR-Global](https://oem-lookup.rbr-global.com/)  and [Sea-Bird](https://instrument.seabirdhub.com/api/argo-calibration?SENSOR_SERIAL_NO=21580&SENSOR_MODEL=SBE41CP&SENSOR_TYPE=) have implemented PREDEPLOYMENT_CALIB_DATE via our Argo JSON metadata APIs.

**Comment by BrianKingNOC** [comment]
I'd like to start a new issue, but I can't find out how to. So I'll start discussion here, and maybe someone can start an issue properly in the right place. Apologies for not being able to figure out how to do it 'correctly. Maybe by username doesn't have authority to create new issues here ?

I intended to raise this at some stage at ADMT-26, but we ran out of time.

The issue is controller board firmware revs, comparable to sensor firmware revs

I believe we have now agreed to a new variable, SENSOR_FIRMWARE_VERSION so that this can be separated from SENSOR_MODEL.

We have the same problem with CONTROLLER_BOARD_TYPE_PRIMARY and associated variables. There is presently an agreed but clumsy fix to put the 'proper' entry for controller board type in the string, and then put the version or anything else inside square brackets after it.

Here are three ideas:

**IDEA 1**

The solution comparable to what we do for sensors would be to put everything that anyone would like to put in the square brackets, into a new variable, CONTROLLER_BOARD_VERSION_PRIMARY

The discussion of JSON at ADMT-26 suggested two other possibilities, if we make the string a JSON object

**IDEA 2**

CONTROLLER_BOARD_TYPE_PRIMARY = {"TYPE":"<board type here>","VERSION":"<other text here>"}

The text to the right of the equals sign is a string. I have omitted the single quotes often used for strings for clarity
This can be decoded cleanly as JSON, and the TYPE must check out against table R28. This is just using JSON to do cleanly what we are already doing. String may need to be longer than the present 32.

They keys are nice, but the variable doesn't really want to be called CONTROLLER_BOARD_TYPE any more, because it contains more than just TYPE. See idea 4 below

**IDEA 3**

Don't use the JSON keys of TYPE and VERSION, but use JSON grammar for the list

CONTROLLER_BOARD_TYPE_PRIMARY = ["<board type here>","<other text here>"]

And require the convention that the first entry is the proper board type. Once again we are letting JSON decoders look after the parsing.



**IDEA 4**

Yes another idea would be to aggregate everything about the controller board into one JSON object

CONTROLLER_BOARD_PRIMARY = {"TYPE":"<board type here>","VERSION":"<other text here>","SERIAL_NO":"<ser no here>"}


If we liked the direction of IDEA 4, we could go the whole way with SENSOR, thus

SENSOR_DEFINITION = {"SENSOR":"CTD_TEMP","SENSOR_MAKER":"SEABIRD","SENSOR_MODEL":"SBE41CP","SENSOR_SERIAL_NO":"14443","SENSOR_FIRMWARE_VERSION":"<firmware here>"}

And PLATFORM FAMILY, TYPE, MAKER etc.

Would be interested to hear views on any of these ideas. Maybe IDEA 4 is a solution looking for a problem that doesn't really exist, and IDEA 1 is the direction we should go in.
